docs/book/v2/quick-start.md

```diff
@@ -0,0 +1,166 @@
+    private $streamFactory;
+
+    public function __construct(ResponseFactoryInterface $responseFactory, StreamFactoryInterface $streamFactory)
+    {
+        $this->responseFactory = $responseFactory;
+        $this->streamFactory   = $streamFactory;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        /** @var Album $album */
+        $album = $request->getAttribute('album');
+
+        $body = $this->streamFactory->createStream('The name of the album is: ' . $album->getName());
+        return $this->responseFactory->createResponse()->withBody($body);
+    }
+}
+```
+
+RequestHandlers resemble a single MVC Controller action, and will be used as the primary application functionality when
+dispatching a request.
+
+### Middleware
+
+```php
+namespace Application\Middleware;
+
+use Application\Repository\AlbumRepositoryInterface;
+use Fig\Http\Message\StatusCodeInterface;
+use Laminas\Router\RouteMatch;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class AlbumFromRouteMiddleware implements MiddlewareInterface
+{
+    /** @var AlbumRepositoryInterface */
+    private $albumRepository;
+    /** @var ResponseFactoryInterface */
+    private $responseFactory;
+
+    public function __construct(AlbumRepositoryInterface $albumRepository, ResponseFactoryInterface $responseFactory)
+    {
+        $this->albumRepository = $albumRepository;
+        $this->responseFactory = $responseFactory;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        /** @var RouteMatch $routeMatch */
+        $routeMatch = $request->getAttribute(RouteMatch::class);
+        $albumId    = $routeMatch->getParam('album_id');
+        $album      = $this->albumRepository->findById($albumId);
+
+        // if no album was found, we short-circuit the pipe and return a 404 error:
+        if ($album === null) {
+            return $this->responseFactory->createResponse(
+                StatusCodeInterface::STATUS_NOT_FOUND,
+                sprintf('Album with ID %s not found!', $albumId)
+            );
+        }
+
+        // ...otherwise we populate the request with the album and call the RequestHandler
+        $request = $request->withAttribute('album', $album);
+        return $handler->handle($request);
+    }
+}
+```
+
+Middleware can return a direct response, in effect short-circuiting the middleware pipe, or pass request further
+while having a chance to act on passed request or returned response.
+Middleware in laminas-mvc is similar to [routed middleware](https://docs.mezzio.dev/mezzio/v3/features/router/piping/#routing)
+in Mezzio. Laminas-mvc does not have a global middleware pipe, so middleware can not be piped in front of MVC controllers.
```

Comment:
May be elevate the note about middleware in front of controllers to a note block to be more prominently visible?